Sources/TensorFlow/Operators/Basic.swift

```diff
@@ -1334,8 +1334,12 @@ extension Tensor {
 
   /// Returns `true` if the given scalar tensor is in the range `[-rank, rank)`.
   @usableFromInline
-  internal func isAxisInRange(_ axis: Tensor<Int32>) -> Bool {
-    precondition(axis.rank == 0, "Axis must have rank 0.")
+  internal func isAxisInRange(
```

Comment:
I realize you didn't add this name, but the `isSomeThingSomeAdjective` style of predicate naming comes from Cocoa and is not really part of the Swift style (even though many people still do it); `someThingIsSomeAdjective` reads much more naturally.  In this case, `axisIsInRange(a)` or `isInRange(axis: a)` would be more appropriate.  Because `Tensor<Int32>` is weak type information, I might opt for the 2nd version, which is consistent with what you did for `ensureValid(axes:)`

Reply:
Follow on: #756 

FWIW: I'm not sure I appreciate the difference between `isAxisInRange(_:)` vs `isInRange(axis:)`; they seem approximately equivalently awkward, and equivalently comprehensible when encountering them. :-D

Reply:
There's a difference in comprehensibility: `isAxisInRange(q)` could be asking about whether some unseen axis is in the range `q`, rather than whether some axis `q` is in some unseen range.  

Regarding awkwardness, we don't say "if is (the) dog white, do something"  we say "if (the) dog is white, do something." 

"is dog pilot Snoopy" is more analogous to what's going on with `isAxisInRange(a)`; "axis/dog" are not serving a purpose in this scenario other than to make up for weak type information, so it should get out of the primary (base) name and clearly indicate the entity to which it applies.

Make sense?